xbmc/FileItem.cpp

```diff
@@ -3578,11 +3578,44 @@ std::string CFileItem::GetBaseMoviePath(bool bUseFolderNames) const
 
       URIUtils::GetParentPath(name2, strMovieName);
     }
+
+    // Remove 'Disc' subdirectory so actual title passed to scraper
+    if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+            CSettings::SETTING_MYVIDEOS_HANDLEDISCN))
+    {
+      size_t iDisc = GetDiscPath(strMovieName);
+      if (iDisc)
+        strMovieName.erase(iDisc + 1);
+    }
   }
 
   return strMovieName;
 }
 
+size_t CFileItem::GetDiscPath(std::string strPath) const
+{
+  // Handle both Disc and Disk spellings
+  size_t iPos = strPath.rfind("/Disc ");
+  if (iPos == std::string::npos)
+    iPos = strPath.rfind("/Disk ");
+
+  if (iPos != std::string::npos)
+  {
+     // Check remainder of path is numeric (eg. Disc 1)
+    size_t iEnd = strPath.find("/", iPos + 1);
```

Comment:
`const size_t ...`

Please remove the type prefix from the variable's name (coding guideline).